pkg/liquidity-source/curve/twocrypto-ng/pool_tracker.go

```diff
@@ -0,0 +1,245 @@
+		Params: nil,
+	}, []interface{}{&midFee})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodOutFee,
+		Params: nil,
+	}, []interface{}{&outFee})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodFutureAGammaTime,
+		Params: nil,
+	}, []interface{}{&futureAGammaTime})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodFutureAGamma,
+		Params: nil,
+	}, []interface{}{&futureAGamma})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodInitialAGammaTime,
+		Params: nil,
+	}, []interface{}{&initialAGammaTime})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodInitialAGamma,
+		Params: nil,
+	}, []interface{}{&initialAGamma})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodLastTimestamp,
+		Params: nil,
+	}, []interface{}{&lastTimestamp})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodXcpProfit,
+		Params: nil,
+	}, []interface{}{&xcpProfit})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodVirtualPrice,
+		Params: nil,
+	}, []interface{}{&virtualPrice})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodAllowedExtraProfit,
+		Params: nil,
+	}, []interface{}{&allowedExtraProfit})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: poolMethodAdjustmentStep,
+		Params: nil,
+	}, []interface{}{&adjustmentStep})
+
+	calls.AddCall(&ethrpc.Call{
+		ABI:    curveTwocryptoNGABI,
+		Target: p.Address,
+		Method: shared.ERC20MethodTotalSupply,
+		Params: nil,
+	}, []interface{}{&lpSupply})
+
+	for i := range p.Tokens {
+		calls.AddCall(&ethrpc.Call{
+			ABI:    curveTwocryptoNGABI,
+			Target: p.Address,
+			Method: poolMethodBalances,
+			Params: []interface{}{big.NewInt(int64(i))},
+		}, []interface{}{&balances[i]})
+	}
+
+	for i := 0; i < numDepCoins; i++ {
+		calls.AddCall(&ethrpc.Call{
+			ABI:    curveTwocryptoNGABI,
+			Target: p.Address,
+			Method: poolMethodPriceScale,
+			Params: nil,
+		}, []interface{}{&priceScales[i]})
+
+		calls.AddCall(&ethrpc.Call{
+			ABI:    curveTwocryptoNGABI,
+			Target: p.Address,
+			Method: poolMethodPriceOracle,
+			Params: nil,
+		}, []interface{}{&priceOracles[i]})
+
+		calls.AddCall(&ethrpc.Call{
+			ABI:    curveTwocryptoNGABI,
+			Target: p.Address,
+			Method: poolMethodLastPrices,
+			Params: nil,
+		}, []interface{}{&lastPrices[i]})
+	}
+	if res, err := calls.TryBlockAndAggregate(); err != nil {
+		lg.WithFields(logger.Fields{"error": err}).Error("failed to aggregate call pool data")
+		return entity.Pool{}, err
+	} else if res.BlockNumber != nil {
```

Comment:
Because in case of error, the code returns immediatelly, I think we can separate the second if block

```suggestion
	}
    
    if res.BlockNumber != nil {
```

Reply:
separating the second if block in this case doesn't reduce the indent level of the nested block and makes the code longer though? moreover in this case, `res` is scoped within the `if`